### PR TITLE
Version support in build and cli

### PR DESCRIPTION
**Pull request by cquinn**
Added VERSION file for use by Makefile.
Passing version string through to Pony code using an emitted source file.
Added new version command to print the version string.

Issue: https://github.com/ponylang/corral/issues/5

Question: is there a way to pass a string value to pony code using `ponyc -D`?

### CODE REVIEW

File: corral/cmd/cmd_version.pony [hunk 1]
```diff
@@ -0,0 +1,8 @@
+use "cli"
+use ".."
+
+primitive CmdVersion
+  fun apply(ctx: Context, cmd: Command) =>
+    //ctx.log.info("version: " + cmd.string())
```
Comment (SeanTAllen) [inline corral/cmd/cmd_version.pony]: you left behind a comment.

**Comment by SeanTAllen** [comment]
I'm not a fan of this approach. I think the "generate version file on the fly from string in Makefile" is worse than the "sed magic" approach that stable takes.

I'd rather do the sed magic approach but beef up the Makefile to handle edge cases better.

@cquinn I'm going to take some of your work here and open a PR with sed-magic and we can compare the two to decide which is "better".

**Comment by SeanTAllen** [comment]
If we go with this approach, we'll need to update the makefile as the created binaries need to have a dependency of VERSION for this to work correctly and rebuild as needed when version changes. See #14  for how that would be done.

**Comment by cquinn** [comment]
I'm happy to take your PR instead of this one. I hadn't looked at the stable Makefile, and that does seem more clear.

**Pull request closed without merging**